x-pack/plugin/ilm/src/test/java/org/elasticsearch/xpack/ilm/action/TransportExplainLifecycleActionTests.java

```diff
@@ -217,6 +225,84 @@ public void testGetIndexLifecycleExplainResponse_rolloverOnlyIfHasDocuments_adds
         assertThat(rolloverAction.getConditions().getMinDocs(), is(1L));
     }
 
+    public void testPreviousStepInfoTruncationDoesNotBreakExplainJson() throws Exception {
+        final String policyName = "policy";
+        final String indexName = "index";
+
+        final String longReasonMessage = "a".repeat(LifecycleExecutionState.MAXIMUM_STEP_INFO_STRING_LENGTH);
+        final String errorJsonWhichWillBeTruncated = Strings.toString((builder, params) -> {
+            ElasticsearchException.generateThrowableXContent(
+                builder,
+                ToXContent.EMPTY_PARAMS,
+                new IllegalArgumentException(longReasonMessage)
+            );
+            return builder;
+        });
```

Comment:
Small nit, but I think we can make this slightly more compact like so:
```suggestion
        final final String errorJsonWhichWillBeTruncated = Strings.toString(new ElasticsearchException(new IllegalArgumentException(longReasonMessage)));
```